Ashrita - Display badges assigned to all users - Badge Management

AshritaCherlapally opened the pull request:
# Description
Created this new PR as the old one https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/3130 was having issues with the version changes.

This PR implements the functionality to display the badges assigned to all users in Badge Management.

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/0e0a158d-95d4-499f-a901-3e07dfc48efb" />

## Related PRS (if any):
This frontend PR is related to the https://github.com/OneCommunityGlobal/HGNRest/pull/1220 PR.

## Main changes explained:
 - When user selects badges from the dropdown list it will display the badges in a Piechart and lists the top 5-10 users assigned to each badge.
- If no users assigned to batch it will display as "No users assigned" and the pie chart won't be displayed for that specific badge.
…

## How to test:
1. check into current branch
2. do npm install and npm run start:local to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ go to OTHER LINKS→ BADGE MANAGEMENT→ Select Badges.
6. Once the badges list is loaded -> Verify Select badges, by selecting multiple badges and ensuring that a pie chart & a list of assigned users(at most 10) is displayed side by side for every badge selected.
7. Also verify if it is working in both light and dark modes.

## Screenshots or videos of changes:
https://github.com/user-attachments/assets/47131f85-c988-435c-9a04-3b53e245e8fe

AshritaCherlapally committed:
Impleted Badge Management Component Pie Chart

rishwa344 commented:
Reviewed the Badge assignment functionality. The system correctly displays "No users assigned" instead of rendering the pie chart. Also confirmed that this behavior remains consistent in both light and dark modes.

<img width="1438" alt="Screenshot 2025-02-28 at 10 41 14 PM" src="https://github.com/user-attachments/assets/7cf8de66-fe43-41d4-8c4c-3127b62d6aed" />

<img width="1435" alt="Screenshot 2025-02-28 at 11 00 28 PM" src="https://github.com/user-attachments/assets/05bb60d1-0a05-4509-b9a0-2924476e16a6" />

barnaboss305 reviewed (changes requested):
When I click on the "select badge" button, it does not show any badges for me to add. Upon further inspection I found that the server was not able to find the elements. Let me know if I am doing anything wrong.

barnaboss305 commented:
> When I click on the "select badge" button, it does not show any badges for me to add. Upon further inspection I found that the server was not able to find the elements. Let me know if I am doing anything wrong.


https://github.com/user-attachments/assets/df28cab5-8301-49a8-b45b-273fd5ffbfad

puchakayalasaisaketh reviewed (commented):
I am not able to find the "select badge" icon.

<img width="468" alt="image" src="https://github.com/user-attachments/assets/1a662e2f-d7cc-48f3-8883-98ea8fda2576" />

kkkaiaa commented:
PR Reviewed. The code works fine in both light and dark mode for me. The pie chart also looks fine if it means showing the two badges that record the longest Working hours and the maximum number of consecutive weeks
<img width="1279" alt="截屏2025-03-01 下午11 11 53" src="https://github.com/user-attachments/assets/a8b838bb-1981-43b1-aab9-d44324b1bf68" />
<img width="1260" alt="截屏2025-03-01 下午11 14 15" src="https://github.com/user-attachments/assets/eb01098c-67b6-4a6d-804a-6140adf2ef91" />

HHHHHeloise commented:
PR Reviewed. 
This code works fine in both light and dark mode for me. 
But it still shows 0 badges selected when there are badges selected. 
Hope there could be a clear selected button otherwise if I selected too many elements it would be difficult for me to find them all and deselect them.
Overall, the functionality is working as expected.
<img width="1245" alt="image" src="https://github.com/user-attachments/assets/681577e6-16ee-4938-b690-05235abf5a1c" />

<img width="1103" alt="badge management 3216" src="https://github.com/user-attachments/assets/337af156-7acb-4559-9934-162e0d21f5f1" />

smohansatyaram reviewed (commented):
The functionality works well for the requirements, but when a large number of badges are selected in the pie chart area, the text alignment tends to get disrupted. It would be helpful to improve the alignment. Additionally, a "Clear Selected" button would be beneficial. Without it, if too many elements are selected, it becomes difficult to locate and deselect them.


![Screenshot 2025-03-18 185129](https://github.com/user-attachments/assets/b9a08045-8082-49ab-9300-f396a9eacee5)
![Screenshot 2025-03-18 185108](https://github.com/user-attachments/assets/b3787cee-6c32-462c-992f-6d3875abd126)

vamsikrishna1704 reviewed (changes requested):
Hi @AshritaCherlapally,
I have reviewed the PR and I have followed the instructions to run the application locally. I have found that Pie chart is not display for many of the badges and also showing the list with no users. So, I would suggest you to recheck the implementation for the pie chart for the user based on the badges assigned. Thank you.
![BadgePieChartNotShowing](https://github.com/user-attachments/assets/38d1c9b0-51b7-4b3f-b432-69c49133fa38)

AshritaCherlapally committed:
Fixed badge selection

AshritaCherlapally committed:
Fixed merge conflict in URL.js

AshritaCherlapally committed:
Resolving merge conflicts

AshritaCherlapally committed:
Resolving the failure

AshritaCherlapally committed:
Fix Prettier formatting issues in badgeReducer.js

AshritaCherlapally commented:
Updated the code as per reviewer suggestions:

1. Added a Clear Selection button.
2. Optimized the loading speed of the badges dropdown.
3.  Limited user selection to a maximum of 10 users for badge review.

https://github.com/user-attachments/assets/9d6aa797-ee8d-4fc5-80bb-61542678715f

Please review and let me know if any further changes are needed.

manushajyasta30 reviewed (approved):
I have cloned both front end and back end PRs.  All the changes mentioned in the PR are working as expected. validated to see if more than 10 badges can be assigned. Good Work.
<img width="959" alt="image" src="https://github.com/user-attachments/assets/c884f668-b13d-4b97-8289-3a52267c3dd0" />

1709abhishek commented on src/components/Badge/BadgeManagement.jsx:
```diff
@@ -46,36 +192,232 @@ function BadgeManagement(props) {
+      >
+        {chartData ? (
+          <div
+            style={{
```
kindly move the inline css

naiduajaykumar reviewed (commented):
I had checkout the both frontend and backend pr and ran it. But when I clicked on "select badges" I don't see any list showing up. No issue with dark theme. Thank you

<img width="1891" height="998" alt="Screenshot 2025-07-10 020040" src="https://github.com/user-attachments/assets/f2664703-b4a7-484a-aa0e-54873634a0aa" />
<img width="1916" height="974" alt="Screenshot 2025-07-10 020101" src="https://github.com/user-attachments/assets/34f43f16-fb5e-4328-a6e1-95618d9808b2" />

BenithaSri reviewed (dismissed):
Followed the steps and I am able to add multiple badged. And it is working in both, light and dark modes.

<img width="609" height="600" alt="Screenshot 2025-07-31 at 1 07 56 AM" src="https://github.com/user-attachments/assets/cff4cff1-0e4d-44e0-b043-c0f99b9e51c6" />
<img width="657" height="602" alt="Screenshot 2025-07-31 at 1 11 34 AM" src="https://github.com/user-attachments/assets/91515077-c4bb-4074-91dd-fc5c2e00b2c7" />

AshritaCherlapally committed:
fix(errors): Resolved merge conflicts

AshritaCherlapally committed:
fix(errors): Resolved merge conflicts

sayali-2308 reviewed (approved):
PR Review - Testing Report

Reviewer: Sayali Sable  
Date: January 27, 2026  
Branch Tested: Ashrita-modified-BadgeManagementComponent  
Backend Tested: muhideen-updated-badges-schema (PR #1220)

Testing Environment
- Frontend: localhost:5173
- Backend: localhost:4500 with PR #1220
- MongoDB: Connected successfully
- Node.js: v20.19.4

Frontend Tests Completed

Badge Selection & Display
- Badge dropdown loads successfully with all available badges
- Selected 6 badges - all display correctly
- Pie chart renders properly with color-coded segments for each badge
- User lists display correctly (showing 5-10 users per badge)

User Assignment Display
- Users properly listed for each selected badge
- "No users assigned" message displays correctly for empty badges (if applicable)
- User names display clearly in the assigned users panel

UI/UX
- "Clear Selected" button works correctly
- Badge selection/deselection updates charts in real-time
- Layout is responsive and professional
- No console errors during normal operation

Dark Mode
- Toggled between light and dark modes
- All components (pie chart, user lists, dropdowns) render correctly in both modes
- Text remains readable in dark mode
- Color contrast is appropriate

Backend Integration
- Frontend successfully calls `/api/badge/getAllBadges`
- Data loads properly from backend
- No 404 or connection errors after configuring local backend URL

Multi-Role Testing
Tested with Admin account (Badge Management requires admin permissions)

Findings
- All functionality works as described
- Frontend-backend integration successful
- No bugs found
- Ready for merge
